src/CovidSim.cpp

```diff
@@ -370,114 +366,126 @@ int main(int argc, char* argv[])
-			fprintf(stderr, "Realisation %i of %i  (time=%lf nr_ne=%i)\n", i + 1, P.NumRealisations,((double)(clock() - cl)) / CLOCKS_PER_SEC, P.NRactNE);
+			P.FitIter++;
+			StopFit=ReadFitIter(FitFile);
+			if (!StopFit)
+			{
+				ReadParams(ParamFile, PreParamFile);
+				if(!P.FixLocalBeta) InitTransmissionCoeffs();
+				sprintf(OutFileBase, "%s.f%i", OutFileBaseF, P.FitIter);
+			}
 		}
-		P.StopCalibration = P.ModelCalibIteration = 0;  // needed for calibration to work for multiple realisations
-		P.PreControlClusterIdHolOffset = 0; // needed for calibration to work for multiple realisations
-		P.PreControlClusterIdCaseThreshold = P.PreControlClusterIdCaseThreshold2; // needed for calibration to work for multiple realisations
-		P.SeedingScaling = 1.0; // needed for calibration to work for multiple realisations
-		///// Set and save seeds
-		if (i == 0 || (P.ResetSeeds && P.KeepSameSeeds))
+		else
+			StopFit = 1;
+		if ((!GotF) || (!StopFit))
 		{
-			P.nextRunSeed1 = P.runSeed1;
-			P.nextRunSeed2 = P.runSeed2;
-		}
-		if (P.ResetSeeds) {
-			//save these seeds to file
-			SaveRandomSeeds();
-		}
-		// Now that we have set P.nextRunSeed* ready for the run, we need to save the values in case
-		// we need to reinitialise the RNG after the run is interrupted.
-		int32_t thisRunSeed1 = P.nextRunSeed1;
-		int32_t thisRunSeed2 = P.nextRunSeed2;
-//		if (i == 0 || P.ResetSeeds)
-			setall(&P.nextRunSeed1, &P.nextRunSeed2);
-
-		///// initialize model (for this realisation).
-		InitModel(i); //passing run number into RunModel so we can save run number in the infection event log: ggilani - 15/10/2014
-		if (P.DoLoadSnapshot) LoadSnapshot();
-		int ModelCalibLoop = 0;
-		while (RunModel(i))
-		{  // has been interrupted to reset holiday time. Note that this currently only happens in the first run, regardless of how many realisations are being run.
-			if ((P.ModelCalibIteration == 10) && (ModelCalibLoop < 3))
+			P.NRactE = P.NRactNE = 0;
+			ResetTimeSeries();
+			for (i = 0; (i < P.NumRealisations) && (P.NRactNE < P.NumNonExtinctRealisations); i++)
 			{
-				thisRunSeed1 = P.nextRunSeed1;
-				thisRunSeed2 = P.nextRunSeed2;
-				setall(&P.nextRunSeed1, &P.nextRunSeed2);
-				P.ModelCalibIteration = 0;  // needed for calibration to work for multiple realisations
-				P.PreControlClusterIdHolOffset = 0; // needed for calibration to work for multiple realisations
-				P.PreControlClusterIdCaseThreshold = P.PreControlClusterIdCaseThreshold2; // needed for calibration to work for multiple realisations
-				P.SeedingScaling = 1.0; // needed for calibration to work for multiple realisations
-				ModelCalibLoop++;
+				if (P.NumRealisations > 1)
+				{
+					sprintf(OutFile, "%s.%i", OutFileBase, i);
+					fprintf(stderr, "Realisation %i of %i  (time=%lf nr_ne=%i)\n", i + 1, P.NumRealisations, ((double)(clock() - cl)) / CLOCKS_PER_SEC, P.NRactNE);
+				}
+				///// Set and save seeds
+				if (((i == 0) && (P.FitIter==1)) || (P.ResetSeeds && P.KeepSameSeeds))
```

Comment:
The regression tests fail because P.nextRunSeed1 and P.nextRunSeed2 do not always get initialised (and the seeds must be non-zero).  I have the following as a suggested fix.  But I am not sure it is correct as I haven't checked that it is only the first initialisation we may miss:

```suggestion
				if (P.nextRunSeed1 == 0 || P.nextRunSeed2 == 0 || ((i == 0) && (P.FitIter==1)) || (P.ResetSeeds && P.KeepSameSeeds))
```

An example command line (based on the us-input data in tests/us-input):

```sh
cp data/populations/wpop_usacan.txt.gz /tmp/pop.txt.gz
gunzip /tmp/pop.txt.gz
./CovidSim /c:1 /BM:bmp /PP:tests/us-input/pre-params.txt /P:tests/us-input/input-noint-params.txt /tmp/st-results-noint /D:/tmp/pop.txt /M:/tmp/st-pop.bin /A:tests/us-input/admin-params.txt /S:/tmp/st-network.bin /R:1.5 98798150 729101 17389101 4797132
```